Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM php:8.1.12-cli
+
+RUN apt-get update && apt-get install -y -q git rake ruby-ronn zlib1g-dev && apt-get clean
+
+RUN cd /usr/local/bin && curl -sS https://getcomposer.org/installer | php
+RUN cd /usr/local/bin && mv composer.phar composer
+RUN pecl install grpc
+RUN docker-php-ext-enable grpc
+
+COPY . /app
+
+WORKDIR /app
```

Comment:
If we copy the app dir into the docker image then we have to re-build it every time we change the app, and also it makes this docker image very tightly coupled to this repo.

Instead I think we can use `-v` flags when we call `docker run`, and that will allow us to mount the appropriate directories into the docker container.  So I think we can get rid of these lines.

Reply:
I think with Dockerfile it might be difficult to achieve this.
What I have tried during docker build:
- Copy only `composer.json` and `composer.lock` 
- Run `composer install` 
- Results in creation of `vender` which includes all the packages' files

And at `docker run`, set a volume to mount client-sdk-php to a Docker container.
But what happens is that this volume overwrites (well dah) `vendor` and now there are no packages installed.

It looks like we could specify which directories/files to mount using `docker-compose.yml`. But I don't think that achieve what we're trying to do, which re-using this image for other purposes.

Reply:
okay, I will pull this down and look into it when i get a chance.  It might be tomorrow.

I suspect we can just mount the vendor directory to keep the image generic but I need to double check to make sure it doesn't contain any native extensions or anything like that.